src/containers/Admin/ShareTab/index.tsx

```diff
@@ -0,0 +1,187 @@
+
+    const [ownersData, setOwnersData] = useState<BoardOwnersData[]>([])
+    const [requestedOwnersData, setRequestedOwnersData] = useState<
+        BoardOwnersData[]
+    >([])
+    const [ownerRequests, setOwnerRequests] = useState<OwnerRequest[]>([])
+    const [boardName, setboardName] = useState<string>('')
+
+    const documentId = getDocumentId()
+
+    const handleDismissLoginModal = (newUser: User | undefined): void => {
+        if (!newUser || newUser.isAnonymous) {
+            setLoginModalOpen(false)
+            setTabIndex(0)
+        }
+    }
+
+    const handleDismissNeedToBeOwnerModal = (goToFirstTab = false): void => {
+        setNeedToBeOwnerModalOpen(false)
+        if (goToFirstTab) setTabIndex(0)
+    }
+
+    useEffect((): void => {
+        if (tabIndex === 5 && user && user.isAnonymous) {
+            setLoginModalOpen(true)
+        }
+
+        if (user && !user.isAnonymous) {
+            if (tabIndex === 5 && !owners.includes(user.uid)) {
+                setNeedToBeOwnerModalOpen(true)
+            } else {
+                setLoginModalOpen(false)
+                setNeedToBeOwnerModalOpen(false)
+            }
+        }
+    }, [user, owners, tabIndex, setTabIndex])
+
+    useEffect(() => {
+        const unsubscribeBoard = getBoardOnSnapshot(documentId, {
+            next: (documentSnapshot: DocumentSnapshot) => {
+                if (!documentSnapshot.exists) return
+                if (documentSnapshot.metadata.hasPendingWrites) return
+                getOwnerEmailsByUIDs(documentSnapshot.data()?.owners).then(
+                    (data) => setOwnersData(data),
+                )
+                getOwnerEmailsByUIDs(
+                    documentSnapshot.data()?.ownerRequestRecipients,
+                ).then((data) => setRequestedOwnersData(data))
+                setOwnerRequests(documentSnapshot.data()?.ownerRequests)
+                setboardName(documentSnapshot.data()?.boardName)
+            },
+            error: () => {
+                setOwnersData([])
+                setRequestedOwnersData([])
+            },
+        })
+
+        return () => unsubscribeBoard()
+    }, [documentId])
+
+    if (!documentId) {
+        return (
+            <div className="share-page">
+                <Heading2 className="heading">Del din tavle med andre</Heading2>
+                <Paragraph className="share-page__paragraph">
+                    Vi har oppgradert tavla. Ønsker du tilgang på denne
+                    funksjonaliteten må du lage en ny tavle.
+                </Paragraph>
+            </div>
+        )
+    }
+
+    return (
+        <div className="share-page">
+            <Heading2 className="heading">Del din tavle med andre</Heading2>
+            <Paragraph>
+                Denne siden lar deg dele den låste tavlen din med andre, slik at
+                dere kan samarbeide på
+                {String.fromCharCode(160)}den.
```

Comment:
Kva er dette fromCharCode-greiene?

Reply:
Hehe, er en non-breaking space. Gjør at det ikke blir ett ord liggende alene på en linje. For å være teknisk unngår det enke (https://no.wikipedia.org/wiki/Enke_(typografi)). Er bare for at det skal se litt bedre ut på de vindusbreddene det gjelder. Kan fjernes hvis du ikke liker den litt skjulte funksjonaliteten i koden

Reply:
Den er grei, men er kanskje vanlegare å bruke `&nbsp;` 😊

Reply:
Fikk ikke til å bruke den først, men skal prøve igjen! 👍